back-end/api/quickstart/tests.py

```diff
@@ -112,3 +112,76 @@ def test_get_valid_post(self):
   def test_get_invalid_post(self):
     response = client.get('/posts/invalidId/')
     self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+
+class UpdatePostById(TestCase):
+  """Tests for updating a single Author by PUT'ing to endpoint /author/_id/.
+  Referenced https://realpython.com/test-driven-development-of-a-django-restful-api/
+  """
+  def setUp(self):
+    self.post = Post.objects.create(
+      _id='123',
+      title='testpost',
+      description='i am a test post',
+      source='source post id',
+      origin='origin post id',
+      visibility='Public',
+      unlisted='True',
+      isPrivateToFriends='True',
+      author='testId',
+      contentType = 'text/plain',
+      content='Hello, I am a test post',
+      categories='["Testing"]',
+      published=datetime.datetime.now(),
+      count=5,
+      pageSize=20,
+      commentLink='link to comments',
+      comments='{ "text": "nice test" }'
+    )
+
+    self.payload = {
+      # TODO: Remove _id here once we figure out the _id field. It should be uneditable.
+      '_id': '123', 
+      'title': 'newpost', 
+      'description': 'i am a new post', 
+      'source': 'new source post id', 
+      'origin': 'new origin post id', 
+      'visibility': 'F', 
+      'unlisted': False, 
+      'isPrivateToFriends': False, 
+      'author': 'newId', 
```

Comment:
this author id should also be uneditable like the _id right? cause it shouldn't change when updating posts

Reply:
Good point, added a comment here to remove this later as well (once fields are figured out).